### PR TITLE
Fix/1160 fix clustering

### LINKED ISSUE #1160: Node doesn't add itself to database if a broken node already in database.
We are running 0.8.0 with a postgress backend, running inside docker on elasticbeanstalk.

It looks like if you have a dead node in your database, when you start a new node, it doesn't register itself with postgres, so then you cannot establish a successful cluster.

So we have a down node is in our database (`172.17.55.221`).

```
kong=> select * from nodes;
           name            | cluster_listening_address |     created_at
---------------------------+---------------------------+---------------------
 bfcf7816291c_0.0.0.0:7946 | 172.17.55.221:7946        | 2016-04-18 19:30:27
(1 row)
```

When we start a new node, we get this (I have obfuscated our database info in this log):

```
[ec2-user@ip-172-17-54-190 ~]$ sudo docker logs 847f6171561e
[INFO] kong 0.8.0
[INFO] Using configuration: /etc/kong/kong.yml
[INFO] Setting working directory to /usr/local/kong
[INFO] database...........postgres host=... database=kong user=.. password=... port=5432
[INFO] dnsmasq............address=127.0.0.1:8053 dnsmasq=true port=8053
[INFO] serf ..............-profile=wan -rpc-addr=127.0.0.1:7373 -event-handler=member-join,member-leave,member-failed,member-update,member-reap,user:kong=/usr/local/kong/serf_event.sh -bind=0.0.0.0:7946 -advertise=172.17.54.190:7946 -node=847f6171561e_0.0.0.0:7946 -log-level=err
[INFO] Trying to auto-join Kong nodes, please wait..
[WARN] Cannot join 172.17.55.221:7946. If the node does not exist anymore it will be automatically purged.
[WARN] Could not join the existing cluster
```

And then I check the postgres database again, notice how `172.17.54.190` did not register itself:

```
kong=> select * from nodes;
           name            | cluster_listening_address |     created_at
---------------------------+---------------------------+---------------------
 bfcf7816291c_0.0.0.0:7946 | 172.17.55.221:7946        | 2016-04-18 19:30:27
(1 row)
```

Then I deleted all of the nodes:

```
kong=> delete from nodes;
DELETE 0
kong=> select * from nodes;
 name | cluster_listening_address | created_at
------+---------------------------+------------
```

And restarted kong again:

```
[ec2-user@ip-172-17-54-190 ~]$ sudo docker logs 036e5bdeaa86
[INFO] kong 0.8.0
[INFO] Using configuration: /etc/kong/kong.yml
[INFO] Setting working directory to /usr/local/kong
[INFO] database...........postgres host=... database=kong user=.. password=... port=5432
[INFO] dnsmasq............address=127.0.0.1:8053 dnsmasq=true port=8053
[INFO] serf ..............-profile=wan -rpc-addr=127.0.0.1:7373 -event-handler=member-join,member-leave,member-failed,member-update,member-reap,user:kong=/usr/local/kong/serf_event.sh -bind=0.0.0.0:7946 -advertise=172.17.54.190:7946 -node=036e5bdeaa86_0.0.0.0:7946 -log-level=err
[INFO] Trying to auto-join Kong nodes, please wait..
[WARN] Cannot auto-join the cluster because no nodes were found
```

Now it properly registered itself:

```
kong=> select * from nodes;
           name            | cluster_listening_address |     created_at
---------------------------+---------------------------+---------------------
 036e5bdeaa86_0.0.0.0:7946 | 172.17.54.190:7946        | 2016-04-18 20:42:22
```

And I restarted the other kong node we had:

```
[ec2-user@ip-172-17-50-141 ~]$ sudo docker logs 68f7bf32313a
[INFO] kong 0.8.0
[INFO] Using configuration: /etc/kong/kong.yml
[INFO] Setting working directory to /usr/local/kong
[INFO] database...........postgres host=... database=kong user=.. password=... port=5432
[INFO] dnsmasq............address=127.0.0.1:8053 dnsmasq=true port=8053
[INFO] serf ..............-profile=wan -rpc-addr=127.0.0.1:7373 -event-handler=member-join,member-leave,member-failed,member-update,member-reap,user:kong=/usr/local/kong/serf_event.sh -bind=0.0.0.0:7946 -advertise=172.17.50.141:7946 -node=68f7bf32313a_0.0.0.0:7946 -log-level=err
[INFO] Trying to auto-join Kong nodes, please wait..
[INFO] Successfully auto-joined 172.17.54.190:7946
```

And that one registered as well:

```
kong=> select * from nodes;
           name            | cluster_listening_address |     created_at
---------------------------+---------------------------+---------------------
 68f7bf32313a_0.0.0.0:7946 | 172.17.50.141:7946        | 2016-04-18 20:45:05
 036e5bdeaa86_0.0.0.0:7946 | 172.17.54.190:7946        | 2016-04-18 20:42:22
(2 rows)
```

### PR DESCRIPTION
**Pull request by asknaskan**
### Summary

I just changed the position of inconsistency check from clustering to prevent that an node removes itself if it failed to cluster.

Instead of checking it when joining an cluster, I now check it when starting the node.

### Full changelog

* Fixed bug that unclustered nodes remove themself from database when failed to join a cluster.

### Issues resolved

Fix #1160

### CODE REVIEW

**Comment by thefosk** [comment]
@asknaskan it seems like there are unresolved conflictsw

**Comment by asknaskan** [comment]
@thefosk Well, I noticed a bit late I'm being at master code state. I created something similar on the next code base. The next version is not working at the vagrant kong, so I have to wait for checks - But it should fit.

**Comment by asknaskan** [comment]
Look's like there is a lot of stuff broken at the moment, so the checks failed. I'm not be able to start an kong on next branch (docker) at the moment. 
The changes are very small and fitting to the old on master branch code I changed.

File: kong/cmd/utils/serf_signals.lua [hunk 1]
```diff
@@ -92,6 +92,9 @@ function _M.start(kong_config, dao)
     return nil, "could not start Serf: "..err
   end
 
+  -- cleanup current node from cluster to prevent inconsistency of data
+  serf:cleanup()
```
Comment (thefosk) [inline kong/cmd/utils/serf_signals.lua]: @asknaskan can you please check for errors here?

```lua
local ok, err = serf:cleanup()
if not ok then return nil, err end
```

**Pull request merged**